src/main/kotlin/com/exactpro/th2/check1/rule/AbstractCheckTask.kt

```diff
@@ -145,7 +151,11 @@ abstract class AbstractCheckTask(
     @Volatile
     protected var started = false
 
-    protected fun createRootEvent() = Event.from(submitTime).description(description)
+    class RuleRootEvent(timestamp: Instant) : Event(timestamp) {
+        val status: Status get() = status
+    }
```

Comment:
I don't quite understand why we need this class. What was wrong with the previous implementation?